In https://github.com/pandas-dev/pandas-ci/pull/6 the CI run https://travis-ci.org/pandas-dev/pandas-ci/jobs/502590460 failed with

```
[pip-test-env before]
# packages in environment at /home/travis/miniconda3/envs/pip-test-env:
#
# Name                    Version                   Build  Channel
ca-certificates           2019.1.23                     0  
certifi                   2018.11.29               py37_0  
libedit                   3.1.20181209         hc058e9b_0  
libffi                    3.2.1                hd88cf55_4  
libgcc-ng                 8.2.0                hdf63c60_1  
libstdcxx-ng              8.2.0                hdf63c60_1  
ncurses                   6.1                  he6710b0_1  
openssl                   1.1.1b               h7b6447c_0  
pip                       19.0.3                   py37_0  
python                    3.7.2                h0371630_0  
readline                  7.0                  h7b6447c_5  
setuptools                40.8.0                   py37_0  
sqlite                    3.26.0               h7b6447c_0  
tk                        8.6.8                hbc83047_0  
wheel                     0.33.1                   py37_0  
xz                        5.2.4                h14c3975_4  
zlib                      1.2.11               h7b6447c_3  
[pip install]
Processing ./dist/pandas-0.25.0.dev0+203.g3e652ac56.tar.gz
    Complete output from command python setup.py egg_info:
    Traceback (most recent call last):
      File "/home/travis/miniconda3/envs/pip-test-env/lib/python3.7/site-packages/pkg_resources/__init__.py", line 359, in get_provider
        module = sys.modules[moduleOrReq]
    KeyError: 'numpy'
    
    During handling of the above exception, another exception occurred:
    
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-req-build-ngo5yevd/setup.py", line 738, in <module>
        ext_modules=maybe_cythonize(extensions, compiler_directives=directives),
      File "/tmp/pip-req-build-ngo5yevd/setup.py", line 481, in maybe_cythonize
        numpy_incl = pkg_resources.resource_filename('numpy', 'core/include')
      File "/home/travis/miniconda3/envs/pip-test-env/lib/python3.7/site-packages/pkg_resources/__init__.py", line 1144, in resource_filename
        return get_provider(package_or_requirement).get_resource_filename(
      File "/home/travis/miniconda3/envs/pip-test-env/lib/python3.7/site-packages/pkg_resources/__init__.py", line 361, in get_provider
        __import__(moduleOrReq)
    ModuleNotFoundError: No module named 'numpy'
    
    ----------------------------------------
Command "python setup.py egg_info" failed with error code 1 in /tmp/pip-req-build-ngo5yevd/
real	0m1.582s
user	0m1.448s
sys	0m0.124s
The command "ci/install.sh" failed and exited with 1 during .
```

Which is good (that was against master).

I'm going to submit a temporary commit there that applies this patch.

--- a/doc/source/whatsnew/v0.24.2.rst
+++ b/doc/source/whatsnew/v0.24.2.rst
@@ -30,6 +30,7 @@ Fixed Regressions
 - Fixed regression in :class:`TimedeltaIndex` where ``np.sum(index)`` incorrectly returned a zero-dimensional object instead of a scalar (:issue:`25282`)
 - Fixed regression in ``IntervalDtype`` construction where passing an incorrect string with 'Interval' as a prefix could result in a ``RecursionError``. (:issue:`25338`)
 - Fixed regression in :class:`Categorical`, where constructing it from a categorical ``Series`` and an explicit ``categories=`` that differed from that in the ``Series`` created an invalid object which could trigger segfaults. (:issue:`25318`)
+- Fixed pip installing from source into an environment without NumPy (:issue:`25193`)
 
 .. _whatsnew_0242.enhancements:
 

